python/lsst/ts/cbp/mock_server.py

```diff
@@ -85,65 +169,205 @@ async def cmd_loop(self, reader, writer):
                             msg = await called_command(parameter) + "\r"
                         msg = msg.encode("ascii")
                         writer.write(msg)
-                        self.log.info(f"Wrote {msg}")
+                        self.log.debug(f"Wrote {msg}")
                         await writer.drain()
                     else:
                         raise Exception("Command is not implemented")
 
-    async def do_azimuth(self, azimuth=None):
-        if azimuth is None:
-            return f"{self.azimuth}"
-        else:
-            self.azimuth = azimuth
+    async def do_azimuth(self):
+        """Return azimuth position.
 
-    async def do_altitude(self, altitude):
-        if altitude is None:
-            return f"{self.altitude}"
-        else:
-            self.altitude = altitude
+        Returns
+        -------
+        str
+        """
+        return f"{self.encoders.AZIMUTH.position()}"
 
-    async def do_focus(self, focus):
-        if focus is None:
-            return f"{self.focus}"
-        else:
-            self.focus = focus
+    async def do_new_azimuth(self, azimuth):
+        """Set the new azimuth position.
 
-    async def do_mask(self, mask):
-        if mask is None:
-            return f"{self.mask}"
-        else:
-            self.mask = mask
+        Parameters
+        ----------
+        azimuth : `float`
 
-    async def do_rotation(self, rotation):
-        if rotation is None:
-            return f"{self.masks[self.mask]}"
-        else:
-            self.masks[self.mask] = rotation
+        Returns
+        -------
+        str
+        """
+        self.encoders.AZIMUTH.set_position(float(azimuth))
+        return ""
+
+    async def do_altitude(self):
+        """Return the altitude position.
+
+        Returns
+        -------
+        str
+        """
+        return f"{self.encoders.ELEVATION.position()}"
+
+    async def do_new_altitude(self, altitude):
+        """Set the new altitude position.
+
+        Parameters
+        ----------
+        altitude : `float`
+
+        Returns
+        -------
+        str
+        """
+        self.encoders.ELEVATION.set_position(float(altitude))
```

Comment:
I'll correct my previous statement. I strongly suggest making your mock controller handle out-of-range positions (at least roughly) the same way as the real low-level controller.

I fervently hope the real low-level controller rejects such commands. If it does not then please explain what it does in the doc string for your mock controller class and emulate that.

And whatever you do, please test it in your unit test of the mock controller!

It is important to handle things like this correctly because your CSC may well end up sending an invalid command at some point (you have logic that rejects invalid positions in the CSC, which is good, but what if some future update means you have a limit wrong or that somebody breaks that code and the limit is not properly applied?).